### PR TITLE
fix(cx-10231): set attribute for trigger element with different label

### PR DESCRIPTION
**Pull request by neelabhnagaich**
## Description
fix as a part of CX-10231

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 1]
```diff
@@ -198,11 +198,18 @@ export namespace MenuOverlay {
...
+            if(this.triggerElement.hasAttribute("ariaexpanded")){
+              this.triggerElement.setAttribute("ariaexpanded", "true");
+            }
           }
         } else {
           this.dispatchMenuClose();
           if (this.triggerElement) {
             this.triggerElement.removeAttribute("aria-expanded");
+
+            if(this.triggerElement.hasAttribute("ariaexpanded")){
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:210]: What is the difference between aria-expanded and ariaexpanded ?

Comment (neelabhnagaich) [reply web-components/src/components/menu-overlay/MenuOverlay.ts:210]: the trigger element can have aria-expanded or ariaexpanded incase of custom webcomponent which takes this as a property, that't why handling the case for this

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 2]
```diff
@@ -115,14 +115,13 @@ export namespace MenuOverlay {
     connectedCallback() {
       super.connectedCallback();
       document.addEventListener("click", this.handleOutsideOverlayClick);
-      document.addEventListener("keydown", this.handleOutsideOverlayKeydown);
+      this.addEventListener("keydown", this.handleOutsideOverlayKeydown);
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:118]: Adding event listener in document is needed as it listens in document level to close the overlay. It means if any of the menu overlay is clicked , the other menu overlay will be closed.

Comment (neelabhnagaich) [reply web-components/src/components/menu-overlay/MenuOverlay.ts:118]: With this change we are targeting to close just the top overlay and not others, for every overlay to close we need to press escape

Comment (niti-owner) [reply web-components/src/components/menu-overlay/MenuOverlay.ts:125]: What about the functionality impact at other places where it is used?

Comment (neelabhnagaich) [reply web-components/src/components/menu-overlay/MenuOverlay.ts]: removed this

**Comment by neelabhnagaich** [comment]
changes done in another PR, duplicate

**Pull request closed without merging**